infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutSearchController.kt

```diff
@@ -0,0 +1,47 @@
+package fi.fta.geoviite.infra.tracklayout
+
+import fi.fta.geoviite.infra.authorization.AUTH_ALL_READ
+import fi.fta.geoviite.infra.common.PublishType
+import fi.fta.geoviite.infra.logging.apiCall
+import fi.fta.geoviite.infra.util.FreeText
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.*
+
+data class TrackLayoutSearchResult(
```

Comment:
Oliko ok määrittää controllereissa dataluokkia? Näitä ei juurikaan muualla nimittäin näkynyt. Tälle controllerille ei siis ole ainakaan toistaiseksi matchaavaa Serviceä.

Reply:
Tämä littyy selkeästi searchiin niin sanoisin että on ok. Toinen mahdollinen paikka olisi `TrackLayout.kt`, jossa meillä on paljon muita tällaisia määrityksiä

Reply:
Paikoin niitä on vedetty suoraan controlleriin, varsinkin jos luokan rooli on (kuten tässäkin) olla vain tuo controllerin paluuarvon rakenne. Ehkä esimerkit poistunut ajan myötä. Jos service/dao kerros tuottaisi näitä suoraan, voisi olla parempi että se sijaitsisi muualla, mutta tällaisessa tämä on ihan OK. Vastaava oli aiemmin ainakin InfraModelControllerissa mutta kun niitä kertyi sinne useampi luokka ja osaa alettiin käyttää myös servicestä, niille tehtiin oma InfraModelApi.kt. 

Itse asiassa TrackLayout.kt:sta olen aika ajoin koittanut raapia erillisempiä kokonaisuuksia irti ettei se kasvaisi liian isoksi, mutta joo yleisesti tuo patterni on ollut että useamman tiedoston käyttämät dataluokat on jossain avoimessa <käsite>.kt:ssa ja vain yhden tiedoston käyttämät voi olla siellä tiedostossa itsessään.